polaris-core/src/main/java/org/apache/polaris/core/storage/StorageLocation.java

```diff
@@ -19,8 +19,9 @@
 package org.apache.polaris.core.storage;
 
 import jakarta.annotation.Nonnull;
-import java.net.URI;
+import org.apache.hadoop.yarn.logaggregation.LogCLIHelpers;
 import org.apache.polaris.core.storage.azure.AzureLocation;
+import org.apache.hadoop.fs.Path;
```

Comment:
Not sure it's worth pulling in a lot of Hadoop dependencies. I prefer to keep Hadoop out of the code base.
There's an alternative though for this kind of issue [here in Nessie](https://github.com/projectnessie/nessie/blob/main/tools/storage/uri/src/main/java/org/projectnessie/storage/uri/StorageUri.java)

Reply:
+1 to avoid Hadoop dependencies

Reply:
Thanks @snazy - I think the Nessie code is a bit more complex than what we require at this time, so I've created code inspired from the Nessie code you've linked. I hope this is reasonable - and thanks for the link!

Reply:
Not sure. Nessie's `StorageUri` was built for this issue, actually to work around the fact that Iceberg allows URIs that do not comply with RFC 2396. Why not use something that's already known to work?

Reply:
I'd also prefer reusing Nessie code in Polaris (we have other copy cases already). Nessie's `StorageUri` may seem to do more than what Polaris needs, but if / when we start dealing with more complex cases (e.g. Generic Table location), the extra functionality will probably be required.

Reply:
Side note: reviewing new code takes effort. Nessie's code already when through substantial reviews.